hardware/battery/battery_management_system_protocol.ksy

```diff
@@ -0,0 +1,246 @@
+meta:
+  id: battery_management_system_protocol
+  title: Communication protocol of smart battery management systems from LLT power
+  license: CC0-1.0
+  ks-version: 0.9
+  endian: be
+  bit-endian: be
+
+doc: |
+  Many modern general purpose BMS include a UART/Bluetooth based communication interface.
+  After sending read requests they respond with various information's about the battery state in
+  a custom binary format.
+
+doc-ref: https://www.lithiumbatterypcb.com/Protocol%20English%20Version.rar
+
+seq:
+  - id: magic_start
+    contents: [0xdd]
+  - id: cmd
+    type: u1
+    -affected-by: 778
+
+  - size: 0
+    if: ofs_body_start < 0 # storing current position
+  - id: body
+    type:
+      switch-on: cmd
+      cases:
+        commands::read.to_i: read_req
+        commands::write.to_i: write_req
+        _: response(cmd)
+  - size: 0
+    if: ofs_body_end < 0 # storing current position
+
+  - id: checksum
+    type: u2
+    doc: |
+      Should be equal to the result from: 0x10000 - sum(body)
+      Where sum() is calculated over the value of every byte individually.
+      Body includes everything besides magic start/end byte, cmd and checksum,
+      so excluding 2 bytes at the beginning and 3 bytes at the end.
+  - id: magic_end
+    contents: [0x77]
+
+instances:
+  ofs_body_start:
+    value: _io.pos
+  ofs_body_end:
+    value: _io.pos
+  checksum_input:
+    -affected-by: 84
+    pos: ofs_body_start
+    size: ofs_body_end - ofs_body_start
+
+enums:
+  commands:
+    0xa5: read
+    0x5a: write
+
+types:
+  read_req:
+    seq:
+      - id: req_cmd
+        type: u1
```

Comment:
is it an enum?

Reply:
Most likely it is, because there is the following docblock on this attribute:

https://github.com/kaitai-io/kaitai_struct_formats/blob/064f90a8c8dd7d2f50a624c7a1632493d79df162/hardware/battery/battery_management_system_protocol.ksy#L61-L65

which means that it supposedly should have the same value logic as shown below:

https://github.com/kaitai-io/kaitai_struct_formats/blob/064f90a8c8dd7d2f50a624c7a1632493d79df162/hardware/battery/battery_management_system_protocol.ksy#L239-L245

And I totally agree that it should be backed by an enum instead of relying on a comment to convey this important information.

Reply:
Yes, it is an enum in theory, but same as already discussed on `cmd` this is affected by https://github.com/kaitai-io/kaitai_struct/issues/778

Reply:
> Yes, it is an enum in theory, but same as already discussed on `cmd` this is affected by [kaitai-io/kaitai_struct#778](https://github.com/kaitai-io/kaitai_struct/issues/778)

OK, but why you don't just add these values to the existing `commands` enum:

https://github.com/kaitai-io/kaitai_struct_formats/blob/064f90a8c8dd7d2f50a624c7a1632493d79df162/hardware/battery/battery_management_system_protocol.ksy#L55-L58

and resolve them here

https://github.com/kaitai-io/kaitai_struct_formats/blob/064f90a8c8dd7d2f50a624c7a1632493d79df162/hardware/battery/battery_management_system_protocol.ksy#L239-L245

the same way you do here?

https://github.com/kaitai-io/kaitai_struct_formats/blob/064f90a8c8dd7d2f50a624c7a1632493d79df162/hardware/battery/battery_management_system_protocol.ksy#L25-L31

Reply:
Because I think that is misleading, you would have two commands in a request, for example: read (0xa5) basic_info (0x03).
I can make another `registers` enum though.